src/instagram.erl

```diff
@@ -0,0 +1,314 @@
+-define(TAG_URL(Tag),                   ?API_URL(["/tags/", Tag])).
+-define(TAG_RECENT_MEDIA_URL(Tag),      ?API_URL(["/tags/", Tag, "/media/recent"])).
+-define(TAG_SEARCH_URL,                 ?API_URL(["/tags/search"])).
+-define(USER_URL(User),                 ?API_URL(["/users/", User])).
+
+%%%===================================================================
+%%% Start/stop API
+%%%===================================================================
+-spec start() -> {ok, pid()} | {error, term()}.
+start() ->
+    {ok, ClientId}      = ocellus:get_env(instagram_client_id),
+    {ok, ClientSecret}  = ocellus:get_env(instagram_client_secret),
+    {ok, RedirectURI}   = ocellus:get_env(instagram_redirect_uri),
+    start(ClientId, ClientSecret, RedirectURI).
+
+-spec start(client_id(), client_secret(), redirect_uri()) ->
+    {ok, pid()} | {error, term()}.
+start(ClientId, ClientSecret, RedirectURI) when is_list(ClientId),
+                                                      is_list(ClientSecret),
+                                                      is_list(RedirectURI) ->
+    Client = gen_oauth2:new_client(ClientId, ClientSecret, RedirectURI,
+                                   ?CLIENT_TOKEN_URL(ClientId, RedirectURI)),
+    instagram_sup:start_child(Client).
+
+-spec start(atom(), client_id(), client_secret(), redirect_uri()) ->
+    {ok, pid()} | {error, term()}.
+start(Name, ClientId, ClientSecret, RedirectURI) when is_atom(Name),
+                                                            is_list(ClientId),
+                                                            is_list(ClientSecret),
+                                                            is_list(RedirectURI) ->
+    Client = gen_oauth2:new_client(ClientId, ClientSecret, RedirectURI,
+                                   ?CLIENT_TOKEN_URL(ClientId, RedirectURI)),
+    instagram_sup:start_child(Name, Client).
+
+-spec stop(pid()) -> ok.
+stop(ServerRef) ->
+    gen_oauth2:stop(ServerRef).
+
+%%%===================================================================
+%%% Authorization API
+%%%===================================================================
+-spec get_authorization_url(pid()) -> {ok, authorization_url()}.
+get_authorization_url(ServerRef) ->
+    gen_oauth2:get_authorization_url(ServerRef).
+
+-spec identity(pid()) -> {ok, atom()} | {error, no_identity}.
+identity(ServerRef) ->
+    gen_oauth2:identity(ServerRef).
+
+-spec set_access_token(pid(), access_token()) ->
+    ok | {error, invalid_request}.
+set_access_token(ServerRef, AccessToken) ->
+    gen_oauth2:set_access_token(ServerRef, AccessToken).
+
+-spec get_access_token(pid()) ->
+    {ok, access_token()}
+  | {error, not_authorized}.
+get_access_token(ServerRef) ->
+    gen_oauth2:get_access_token(ServerRef).
+
+%%%===================================================================
+%%% User endpoints API
+%%%===================================================================
+-spec get_user(pid()) ->
```

Comment:
That's a good point. However, I look at it as two things:
1) A minimal set of consistent semantics related to a user, its timeline, and its graph (directed or undirected). These concepts are pretty much universal, so here I think it makes sense to consolidate consistent naming. If need be, we can expose synonyms for these end points;
2) Network-specific end points, which are likely disjoint across networks. These we should preserve the provider's naming.

For my purposes I'm more interested in 1 than 2 (at least right now).